File: server/led.py
from gpiozero import PWMLED


class LED:
    def __init__(self, pin, name):
        self.pin = pin
        self.name = name
        self.led = PWMLED(pin)
        self.setting = 'off'
        self.off()

    def on(self):
        self.led.on()
        self.setting = 'on'

    def off(self):
        self.led.off()
        self.setting = 'off'

    def blink(self):
        self.led.blink()
        self.setting = 'blink'

    def pulse(self):
        self.led.pulse()
        self.setting = 'pulse'

    def pwm(self, val):
        self.led.value = val
        self.setting = 'pwm'

File: client/main/elements/Led.js
import * as api from '../services/ledApi'

function Led(state, setState) {

    function on() {
        api.on(state.pin);
        setState({...state, setting: "on"})
    }

    function off() {
        api.off(state.pin);
        setState({...state, setting: "off"})
    }

    function blink() {
        api.blink(state.pin);
        setState({...state, setting: "blink"})
    }

    function pulse() {
        api.pulse(state.pin);
        setState({...state, setting: "pulse"})
    }

    function setValue(val) {
        api.pwm(state.pin, val);
        setState({...state, value: val})
    }

    return {
        name: () => state.name,
        pin: () => state.pin,
        setting: () => state.setting,
        value: () => state.value,
        on: on,
        off: off,
        blink: blink,
        pulse: pulse,
        setValue: setValue,
    }
}

export default Led